### PR TITLE
CIF-925 - Deep reading of properties for GraphQL resource provider

### PR DESCRIPTION
**Pull request by LSantha**
* implemented feature
 * updated unit tests

## Description

As a developer I want support for deep reading of resource properties for GraphQL catalog resources so that category and product resource properties can be referred to as ./propertyName in resource property dialogs and product scaffolding.
This feature is available for the JCR resource provider in AEM and various sling components take it as granted, so it's important for the GraphQL resource provider to fill this gap.

## Related Issue
CIF-925

## How Has This Been Tested?
* manually
* unit test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/resource/GraphqlResourceProvider.java [hunk 1]
```diff
@@ -53,6 +53,11 @@
     public Resource getResource(ResolveContext<T> ctx, String path, ResourceContext resourceContext, Resource parent) {
         LOGGER.debug("getResource called for " + path);
 
+        // needed for testing, harmless during runtime
+        if (path.endsWith("/.")) {
```
Comment (cjelger) [inline bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/resource/GraphqlResourceProvider.java]: Is this really needed? Looks odd to me to check for a path *ending* with `/.` (while your PR fixes property names *starting* with `./`.

Comment (herzog31) [reply bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/graphql/resource/GraphqlResourceProvider.java]: I agree. Mixing production and test code is a bad practice and should be avoided.

File: bundles/cif-connector-graphql/src/test/java/com/adobe/cq/commerce/graphql/resource/GraphqlResourceProviderTest.java [hunk 2]
```diff
@@ -122,11 +124,22 @@ public void setUp() throws Exception {
...
 
         scheduler = mock(Scheduler.class);
         ScheduleOptions opts = mock(ScheduleOptions.class);
         when(scheduler.schedule(any(), any())).thenReturn(Boolean.FALSE);
         when(scheduler.NOW(Mockito.anyInt(), Mockito.anyLong())).thenReturn(opts);
+
+        provider = new GraphqlResourceProvider<>(CATALOG_ROOT_PATH, dataService, scheduler);
+        when(resourceResolver.getResource(any())).then(invocationOnMock -> {
+            String path = (String) invocationOnMock.getArguments()[0];
+            if (path.endsWith("/.")) {
```
Comment (cjelger) [inline bundles/cif-connector-graphql/src/test/java/com/adobe/cq/commerce/graphql/resource/GraphqlResourceProviderTest.java:137]: Thx for moving this to the test code. Just to clarify, why do we need that? It looks odd to me that a path would end with `/.`. Thx.

Comment (LSantha) [reply bundles/cif-connector-graphql/src/test/java/com/adobe/cq/commerce/graphql/resource/GraphqlResourceProviderTest.java:137]: DeepReadValueMapDecorator tries to resolve a path ending in "/."

**Review by herzog31 — APPROVED** [review]

**Review by cjelger — APPROVED** [review]

**Pull request merged**